### PR TITLE
Expose available UIA patterns in developer info

### LINKED ISSUE #5712: include a list of supported UIA patterns into the developer info
hi guys,

inspired by the jaws touch cursor's object properies dialog, I wrote some code to generate a list of supported UIA patterns for the current navigator object. IMHO it would make sense to include such a patterns list into the developer info generated by nvda+f1. 

To use my global plugin, press nvda+shift+f1. If the navigator is placed on a uia object, the patterns list is displayed as a flash message and it's also written to the nvda log (as an info istring). If the navigator isn't placed onto a uia object, an error message is displayed.

I've tested the global plugin with various (non)UIA objects in Windows 7 x64 as well as in Kaspersky Internet Security 2016.

You can get the global plugin from my dropbox: https://dl.dropboxusercontent.com/u/67471/uia_patterns.py

happy testing

Dave

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Closes #5712 

### Summary of the issue:
As @dave090679, it would be helpful to have available UIA patterns in the developer info for easier inspection of available UIA patterns.

### Description of how this pull request fixes the issue:
This is based on @dave090679's proposal, but it also does some things a bit different.

1. It evaluates all UIAHandler.UIA_Is*PatternAvailablePropertyId constants on the UIAHandler. This means that this code doesn't have to be updated when new patterns become available.
2. Rather than fetching every property in a separate call, this uses UIA._prefetchUIACacheForPropertyIDs to request all properties in one cache request.

### Testing performed:
Tested fetching developer info on Windows 10 X64 and in a Win7x86 vm. See also the discussion with @josephsl in #5712. Concerns regarding compatibility with older versions of Windows 10 should be taken care of.

### Known issues with pull request:
None

### Example output
* A list item in Windows Explorer
    + UIA patterns available: LegacyIAccessiblePattern, InvokePattern, ScrollItemPattern, SelectionItemPattern, ValuePattern
* A table item in the Outlook 2016 inbox
    + UIA patterns available: LegacyIAccessiblePattern, GridItemPattern, InvokePattern, ScrollItemPattern, SelectionItemPattern, ValuePattern
* The wifi toggle button in Windows 10 wifi settings
    + UIA patterns available: LegacyIAccessiblePattern, ScrollItemPattern, TogglePattern

### Change log entry:
+ Changes for developers:
    * The developer info for UIA objects now contains a list of the UIA patterns available. (#5712)

### CODE REVIEW

File: source/NVDAObjects/UIA/__init__.py [hunk 1]
```diff
@@ -712,7 +712,11 @@ def _prefetchUIACacheForPropertyIDs(self,IDs):
 			return
 		cacheRequest=UIAHandler.handler.clientObject.createCacheRequest()
 		for ID in IDs:
-			cacheRequest.addProperty(ID)
+			try:
+				cacheRequest.addProperty(ID)
+			except COMError:
+				log.debug("Couldn't add property ID %d to cache request, most likely unsupported on this version of Windows"%ID)
+				continue
```
Comment (josephsl) [inline source/NVDAObjects/UIA/__init__.py]: I think you can remove this line, as after catching this exception, the for loop will continue anyway. Thanks.

**Review by josephsl — APPROVED** [review]
Hi,

Looks fine to me (at least). Thanks for addressing review actions.

**Review by feerrenrut — APPROVED** [review]
Could you please add some example output to the description of this PR. Otherwise this looks good.

**Comment by leonardder** [comment]
Edited the description as requested.

**Pull request merged**